pyproject.toml

```diff
@@ -1,5 +1,6 @@
 [build-system]
-requires = ["setuptools", "wheel"]
+requires = ["setuptools>=44", "wheel"]
```

Comment:
Usually, the build frontend creates a fresh isolated env which gets the latest versions of these deps. So it's not really recommended to limit versions.

Reply:
I personally prefer specifying the lower bound, in fact. And AFAIK there is not established recommendations either way. It is true that existing tools prefer to populate the environment with the latest version, but that is by no means codified, and in my head I can easily justify to myself a tool preferring a cached wheel to populate (to avoid network traffic, for example).

With that said, the choice of version 44 does seem quite arbitrary to me. Setuptools introduced PEP 517 support in 36.6, the compatibility `__legacy__` handler in 40.8, and `backend-path` (the latest PEP update, which pip does not use) in 43.0. 36.6 would be my choice if we are to choose a lower bound.

Reply:
I agree 36.6 is correct (the oldest version that has the features we need).

Relying on features only present in later versions of setuptools (and hence using a higher limit) has a potential impact on who can install pip (although it's going to be very rare edge cases that are affected) and in theory we shouldn't do that without discussion of the trade-offs.

Note: I remain -1 on bothering with this change at all.

Reply:
True. But I recall proposing lower bound when contributing somewhere (setuptools docs or so) and somebody told me that it shouldn't be recommended.

Reply:
Thanks, fixed.